src/android/toga_android/widgets/_pickers.py

```diff
@@ -0,0 +1,113 @@
+from ..libs.android.view import OnClickListener, View__MeasureSpec
+from ..libs.android.widget import EditText
+from .base import Widget
+
+
+class TogaPickerClickListener(OnClickListener):
+    def __init__(self, picker_impl):
+        super().__init__()
+        self.picker_impl = picker_impl
+
+    def onClick(self, _):
+        self.picker_impl._create_dialog()
+
+
+class TogaPickerSetListener:
+    def __init__(self, picker_impl):
+        super().__init__()
+        self.picker_impl = picker_impl
+
+    def listen(self, _, *args):
+        new_value = self.picker_impl._to_obj_converter_class(*self.picker_impl._value_pack_fn(*args))
+
+        self.picker_impl._showing = False
+        self.picker_impl._dialog = None
+        self.picker_impl.interface.value = new_value
+        if self.picker_impl.interface.on_change:
+            self.picker_impl.interface.on_change(self.picker_impl)
+
+    onDateSet = listen
+    onTimeSet = listen
+
+
+class PickerBase(Widget):
+    _icon = None
+    _hint = None
+    _to_obj_converter_class = None
+    _to_str_converter_kwargs = None
+    _dialog_class = None
+    _update_dialog_name = None
+    _value_unpack_fn = None
+    _value_pack_fn = None
+    _extra_dialog_setters = None
+    _extra_dialog_args = None
+    _dialog_listener_class = None
+
+    def __init__(self, interface):
+        super().__init__(interface)
+        if None in (
+            self._icon,
+            self._hint,
+            self._to_obj_converter_class,
+            self._to_str_converter_kwargs,
+            self._dialog_class,
+            self._update_dialog_name,
+            self._value_unpack_fn,
+            self._value_pack_fn,
+            self._extra_dialog_setters,
+            self._extra_dialog_args,
+            self._dialog_listener_class
+        ):
+            raise ValueError("You have to subclass the picker class and define some fields!")
```

Comment:
The intention here is good; but this is an internal class that won't be exposed to the end-user; this kind of safety check isn't required.